### ISSUE TITLE
Supersets Trino integration not able to list table schemas

### ISSUE THREAD

**Issue by EMCP** [apache/superset#21745]
I can successfully connect using both the `hive` and `ubuntu` user in the connection string.. however when I go to create a dataset.. no tables or schemas are being listed.. with either user

I can confirm the tables exist, I can query it in the SQL Lab.. I just cannot proceed from there.

#### How to reproduce the bug

Add a Trino DB with a catalog in the connection string
Once connected try to add a dataset with a known existing table

### Expected results

it should allow me to make a dataset so I can create charts

### Actual results

its blank and you cannot proceed

#### Screenshots

![image](https://user-images.githubusercontent.com/3691722/194712937-bb996b78-bd99-4fc9-a1f8-29231e55103e.png)

### Environment

```
Superset 2.0.0
TRINO 399
HIVE + S3 + Parquet
pypi trino==0.315.0 (I cannot connect with latest 0.317.0 )
```

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ X] I have reproduced the issue with at least the latest released version of superset.
- [ X] I have checked the issue tracker for the same issue and I haven't found one similar.

StackTrace when clicking in the screenshotted area

````
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ubuntu/.local/lib/python3.8/site-packages/superset/models/core.py", line 606, in get_all_view_names_in_schema
    views = self.db_engine_spec.get_view_names(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/superset/db_engine_specs/trino.py", line 103, in get_view_names
    return BaseEngineSpec.get_view_names(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/superset/db_engine_specs/base.py", line 1007, in get_view_names
    views = inspector.get_view_names(schema)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/reflection.py", line 325, in get_view_names
    return self.dialect.get_view_names(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/trino/sqlalchemy/dialect.py", line 214, in get_view_names
    res = connection.execute(sql.text(query), schema=schema)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 2235, in execute
    return connection.execute(statement, *multiparams, **params)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1011, in execute
    return meth(self, multiparams, params)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 298, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1124, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1316, in _execute_context
    self._handle_dbapi_exception(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1510, in _handle_dbapi_exception
    util.raise_(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/trino/sqlalchemy/dialect.py", line 333, in do_execute
    cursor.execute(statement, parameters)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/trino/dbapi.py", line 460, in execute
    added_prepare_header = self._prepare_statement(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/trino/dbapi.py", line 339, in _prepare_statement
    raise trino.exceptions.FailedToObtainAddedPrepareHeader
sqlalchemy.exc.DBAPIError: (trino.exceptions.FailedToObtainAddedPrepareHeader) 
[SQL: SELECT "table_name"
FROM "information_schema"."views"
WHERE "table_schema" = ?]
[parameters: ('stonks',)]
(Background on this error at: http://sqlalche.me/e/13/dbapi)
2022-10-08 16:48:57,519:WARNING:superset.models.core:Get all view names failed
Traceback (most recent call last):
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/trino/sqlalchemy/dialect.py", line 333, in do_execute
    cursor.execute(statement, parameters)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/trino/dbapi.py", line 460, in execute
    added_prepare_header = self._prepare_statement(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/trino/dbapi.py", line 339, in _prepare_statement
    raise trino.exceptions.FailedToObtainAddedPrepareHeader
trino.exceptions.FailedToObtainAddedPrepareHeader

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ubuntu/.local/lib/python3.8/site-packages/superset/models/core.py", line 606, in get_all_view_names_in_schema
    views = self.db_engine_spec.get_view_names(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/superset/db_engine_specs/trino.py", line 103, in get_view_names
    return BaseEngineSpec.get_view_names(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/superset/db_engine_specs/base.py", line 1007, in get_view_names
    views = inspector.get_view_names(schema)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/reflection.py", line 325, in get_view_names
    return self.dialect.get_view_names(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/trino/sqlalchemy/dialect.py", line 214, in get_view_names
    res = connection.execute(sql.text(query), schema=schema)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 2235, in execute
    return connection.execute(statement, *multiparams, **params)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1011, in execute
    return meth(self, multiparams, params)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 298, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1124, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1316, in _execute_context
    self._handle_dbapi_exception(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1510, in _handle_dbapi_exception
    util.raise_(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/home/ubuntu/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/trino/sqlalchemy/dialect.py", line 333, in do_execute
    cursor.execute(statement, parameters)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/trino/dbapi.py", line 460, in execute
    added_prepare_header = self._prepare_statement(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/trino/dbapi.py", line 339, in _prepare_statement
    raise trino.exceptions.FailedToObtainAddedPrepareHeader
sqlalchemy.exc.DBAPIError: (trino.exceptions.FailedToObtainAddedPrepareHeader) 
[SQL: SELECT "table_name"
FROM "information_schema"."views"
WHERE "table_schema" = ?]
[parameters: ('stonks',)]
(Background on this error at: http://sqlalche.me/e/13/dbapi)

```

**Comment by EMCP** [comment]
this error appears fixed by sticking to `trino==0.318.0` pypi module

**Comment by LucaSoato** [comment]
Thanks guys, you've saved me
